### PR TITLE
Changing hardcoded master/develop regex

### PR DESCRIPTION
**Pull request by pwiens**
I encountered an issue where in a repository without a master or develop branch if we change the regex for master and develop to point to branches which do exist (and are used as master and develop branches) if there are no tags in the repo gitversion throws the following

```
  INFO [07/12/17 12:06:02:30] End: Getting branches containing the commit 'fa406c74e8eea6aa5a01cee62acc7f5ce1f6416e'. (Took: 3.49ms)
  INFO [07/12/17 12:06:02:30] Found possible parent branches:
INFO [07/12/17 12:06:02:30] End: Attempting to inherit branch configuration from parent branch (Took: 68.00ms)
ERROR [07/12/17 12:06:02:30] An unexpected error occurred:
System.InvalidOperationException: Could not find a 'develop' or 'master' branch, neither locally nor remotely.
   at GitVersion.BranchConfigurationCalculator.InheritBranchConfiguration(GitVersionContext context, Branch targetBranch, BranchConfig branchConfiguration, IList`1 excludedInheritBranches)
   at GitVersion.BranchConfigurationCalculator.GetBranchConfiguration(GitVersionContext context, Branch targetBranch, IList`1 excludedInheritBranches)
   at GitVersion.GitVersionContext.CalculateEffectiveConfiguration()
   at GitVersion.GitVersionContext..ctor(IRepository repository, Branch currentBranch, Config configuration, Boolean onlyEvaluateTrackedBranches, String commitId)
   at GitVersion.ExecuteCore.<>c__DisplayClass6_0.<ExecuteInternal>b__0(IRepository repo)
   at GitVersion.GitPreparer.WithRepository[TResult](Func`2 action)
   at GitVersion.ExecuteCore.ExecuteGitVersion(String targetUrl, String dynamicRepositoryLocation, Authentication authentication, String targetBranch, Boolean noFetch, String workingDirectory, String commitId, Config overrideConfig, Boolean noCache)
   at GitVersion.SpecifiedArgumentRunner.Run(Arguments arguments, IFileSystem fileSystem)
   at GitVersion.Program.VerifyArgumentsAndRun()
```

This fix changes the fallback to use the regex in the config instead of using a hardcoded regex.

### CODE REVIEW

File: src/GitVersionCore/BranchConfigurationCalculator.cs [hunk 1]
```diff
@@ -109,8 +109,8 @@ static BranchConfig InheritBranchConfiguration(GitVersionContext context, Branch
                 else
                     errorMessage = "Failed to inherit Increment branch configuration, ended up with: " + string.Join(", ", possibleParents.Select(p => p.FriendlyName));
 
-                var chosenBranch = repository.Branches.FirstOrDefault(b => Regex.IsMatch(b.FriendlyName, "^develop", RegexOptions.IgnoreCase)
-                                                                           || Regex.IsMatch(b.FriendlyName, "master$", RegexOptions.IgnoreCase));
+                var chosenBranch = repository.Branches.FirstOrDefault(b => Regex.IsMatch(b.FriendlyName, config.Branches[ConfigurationProvider.DevelopBranchKey].Regex, RegexOptions.IgnoreCase)
```
Comment (asbjornu) [inline src/GitVersionCore/BranchConfigurationCalculator.cs]: Can you please extract this to a `developBranchRegex` variable?

File: src/GitVersionCore/BranchConfigurationCalculator.cs [hunk 2]
```diff
@@ -109,8 +109,8 @@ static BranchConfig InheritBranchConfiguration(GitVersionContext context, Branch
...
+                                                                           || Regex.IsMatch(b.FriendlyName, config.Branches[ConfigurationProvider.MasterBranchKey].Regex, RegexOptions.IgnoreCase));
```
Comment (asbjornu) [inline src/GitVersionCore/BranchConfigurationCalculator.cs]: Can you please extract this to a `masterBranchRegex` variable?

File: src/GitVersionCore/BranchConfigurationCalculator.cs (hunk 1 above)
Comment (pwiens) [inline src/GitVersionCore/BranchConfigurationCalculator.cs]: Done.

File: src/GitVersionCore/BranchConfigurationCalculator.cs (hunk 2 above)
Comment (pwiens) [inline src/GitVersionCore/BranchConfigurationCalculator.cs]: Let me know if what I did isn't what you meant.

**Comment by pwiens** [comment]
Thanks! What nuget package will this be part of?

**Pull request merged**